### PR TITLE
Removing cover letter being mandatory

### PR DESCRIPTION
**Pull request by discodavey**
@Cooryd Can you have a quick look and see if I have removed what I need to remove for cover letter not to be mandatory. I think I have removed everything but would like for you to check and advise.

### CODE REVIEW

**Comment by discodavey** [comment]
@Cooryd As discussed with the other ticket for cookiebot you mentioned there is an issue in the CI which you need to look at and fix.

**Comment by Cooryd** [comment]
Happy with changes but this should only be deployed with a corresponding update to [`reviewer-submission`](https://github.com/libero/reviewer-submission/blob/2f945b7bd1de349cb5a38360a86bc5cc20bd20d4/src/domain/submission/services/models/files-schema.ts#L5).

**Comment by discodavey** [comment]
Have added the content changes to this PR as well as it makes sense to. #6812

**Comment by discodavey** [comment]
Updated link for author guide is also in this PR as well as can go through at the same time.

**Comment by discodavey** [comment]
@NuclearRedeye Looks all ok to go just need the last bit for the browser tests to be approved. If you can just have a quick eye over and then we can push this one live.

**Comment by NuclearRedeye** [comment]
@discodavey did you see @Cooryd comment about the change in reviewer-submission? I can't see a PR, can you look into that.

**Comment by discodavey** [comment]
> @discodavey did you see @Cooryd comment about the change in reviewer-submission? I can't see a PR, can you look into that.

Yes I saw this. I am assuming I just need to remove this?

**Comment by Cooryd** [comment]
https://github.com/libero/reviewer-submission/pull/270 is the corresponding change to `reviewer-submission`, to make cover letter truely optional on server I had to add `.allow('').optional()`. The back end uses a slightly different lib for validation to the front end, can we check for sure that client still works as expected without also adding something similar?

File: tests/browser-tests/Validation.browser.ts [hunk 1]
```diff
@@ -32,12 +32,8 @@ test('files page', async () => {
     const filesPage = new FilesPage();
     await filesPage.assertOnPage();
     await filesPage.next(true);
-    const validationHelper = new ValidationHelper();
-    await validationHelper.assertNumberOfErrors(1);
-    await validationHelper.assertErrorMessage('.cover-letter', 'Please write or paste in your cover letter');
     await filesPage.assertManuscriptUploadError('Please upload a manuscript');
     await filesPage.populateMinimalFields();
-    await validationHelper.assertNumberOfErrors(0);
```
Comment (NuclearRedeye) [inline tests/browser-tests/Validation.browser.ts:40]: Just trying to wrap my head around if this final assertion is needed or not.

**Review by NuclearRedeye — COMMENTED** [review]
Looks OK to me, just one minor comment around the test.

**Comment by Cooryd** [comment]
https://github.com/libero/reviewer-submission/pull/272 has been merged so this should be good to go once reviewed

**Review by NuclearRedeye — APPROVED** [review]

**Pull request merged**